scripts/package_ota.py

```diff
@@ -16,6 +17,48 @@
 AGNOS_UPDATE_URL = os.getenv("AGNOS_UPDATE_URL", "https://commadist.azureedge.net/agnosupdate")
 AGNOS_STAGING_UPDATE_URL = os.getenv("AGNOS_STAGING_UPDATE_URL", "https://commadist.azureedge.net/agnosupdate-staging")
 
+GPT = namedtuple('GPT', ['lun', 'name', 'path', 'start_sector', 'num_sectors', 'ondevice_hash', 'has_ab', 'ota', 'full_check'])
+GPTS = [
+  GPT(0, 'gpt_main_0', FIRMWARE_DIR / 'gpt_main_0.bin', 0, 6, '8928A31FD9EE20F8703649F89833EBA9B55E84B6415E67799C777B163C95A0BD', False, False, True),
+  GPT(1, 'gpt_main_1', FIRMWARE_DIR / 'gpt_main_1.bin', 0, 6, 'FE8EF7653DB588D7420A625920CA06927DFCB0ED8AFF3E3A1C74A52A24398BA6', False, False, True),
+  GPT(2, 'gpt_main_2', FIRMWARE_DIR / 'gpt_main_2.bin', 0, 6, '5CCFC7240C8CBFA2F1A018A2E376CF274A6BAF858C9BFE71951D8E28CAB53C21', False, False, True),
+  GPT(3, 'gpt_main_3', FIRMWARE_DIR / 'gpt_main_3.bin', 0, 6, 'C707979FA21E89519328F4F30C2B21C9C453401CA8303F914C1873D410A95159', False, False, True),
+  GPT(4, 'gpt_main_4', FIRMWARE_DIR / 'gpt_main_4.bin', 0, 6, 'E9405DCD785DBE79412184E1894A9C51AB7DEB33BB612166C4C42A3D2BF42A0E', False, False, True),
+  GPT(5, 'gpt_main_5', FIRMWARE_DIR / 'gpt_main_5.bin', 0, 6, '21AE965F05B2FA8D02E04F1EB74718F9779864F6EACDEB859757D6435E8CCCE3', False, False, True),
+]
+
+Partition = namedtuple('Partition', ['name', 'path', 'ondevice_hash', 'has_ab', 'ota', 'full_check'])
+PARTITIONS = [
+  Partition('persist', FIRMWARE_DIR / 'persist.bin', '9814B07851292F510F3794B767489F38AB379A99F0EA75DC620AD2D3A496D54D', False, False, True),
```

Comment:
why can't we calculate these hashes on the fly?

Reply:
We could, be we would have to duplicate the logic to handle sparse image + the logic for when the image file is not a multiple of 4096 (since those hashes represent what we actually write on the UFS)